app/lib/clients/vault/certificate.rb

```diff
@@ -15,8 +16,20 @@ def configure_pki
         enable_ca
         sign_cert
         configure_ca
+        create_generic_cert_policy
       end
 
+      def config_user(identity)
+        sub = identity.sub
+        email = identity.email
+        policies, metadata = get_entity_data(sub)
+        policies.append(Certificate::GENERIC_CERT_POLICY_NAME).to_set.to_a
+        put_entity(sub, policies, metadata)
+        put_entity_alias(sub, email, "oidc")
+      end
+
+      GENERIC_CERT_POLICY_NAME = "astral-generic-cert-policy"
```

Comment:
would this give all users the same policy for certs? Once granted, could I see any astral-created cert, regardless of user?

Reply:
>  would this give all users the same policy for certs?

yes, it would allow any user with the policy to generate astral certs.

> Once granted, could I see any astral-created cert, regardless of user?

Regardless of this policy, the read cert api endpoint is unauthenticated so anyone can read the certs, but not recover the private key, (from https://developer.hashicorp.com/vault/api-docs/secret/pki#read-certificate)

The private key is not stored. If you do not save the private key from the response, you will need to request a new certificate (from https://developer.hashicorp.com/vault/api-docs/secret/pki#generate-certificate-and-key )